obspy/io/seg2/tests/test_seg2.py

```diff
@@ -95,7 +114,31 @@ def test_read_data_format_2(self):
                                         atol=1e-7))
         # test seg2 specific header values
         # (trace headers include SEG2 file header)
-        self.assertEqual(st[0].stats.seg2, TRACE1_HEADER)
+        self.assertEqual(st[0].stats.seg2, TRACE2_HEADER)
+
+    def test_read_data_format_3(self):
+        """
+        Test reading a SEG2 data format code 3 file (20-bit floating point).
+        """
+        basename = os.path.join(self.path,
+                                '20180307_031245000.0')
+        # read SEG2 data (in counts, int32)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
+            st = read(basename + ".seg2")
+        self.assertEqual(len(w), 2)
+        self.assertIn('custom', str(w[1]))
+        # read reference ASCII data
+        with gzip.open(basename + '.DAT.gz', 'rb') as f:
+            results = np.loadtxt(f, ndmin=2).T
+        for tr, result in zip(st, results):
+            # convert raw data to unit'd
+            scaled_data = tr.data * tr.stats.calib
```

Comment:
How do we in other formats actually deal with the scaling factor? Do we want it applied directly or not?

Reply:
I'm not sure; when I plot it, the calibration is definitely applied somewhere. I just copied this line from the other test (but kept out the mm/um unit conversion, since I wasn't sure what units these were). It could be the other test is wrong, though.

Maybe @stefanazz could clarify the exact units and expected magnitude of the `1068.DAT` file he provided.

Reply:
Yea you are right - I just looked through a couple of other modules and we do not directly apply the calibration factor upon reading so what you are doing here is the correct thing.

Don't know about the unit conversion though.